cpp/include/kvikio/defaults.hpp

```diff
@@ -57,7 +57,13 @@ inline bool getenv_or(std::string_view env_var_name, bool default_val)
   }
   // Convert to lowercase
   std::string str{env_val};
-  std::transform(str.begin(), str.end(), str.begin(), ::tolower);
+  // Special considerations regarding the case conversion:
+  // - std::tolower() is not an addressable function. Passing it to std::transform() as
+  //   a function pointer causes the program behavior "unspecified (possibly ill-formed)",
```

Comment:
does this point need to be removed? based on https://github.com/rapidsai/kvikio/pull/493/files#r1795596566

Reply:
We are using std::tolower in this PR, so I think this comment is correct.

Reply:
I've expanded the comment a bit more to clarify.